### PR TITLE
electron reco modifications for 2018 heavy ion era

### PR DESCRIPTION
**Pull request by bi-ran**
this PR includes the following modifications to electron reconstruction for the 2018 heavy ion era:
- raise supercluster energy thresholds
- removal of tracker-driven electrons

these changes affect mainly very low-pt electrons, and are in line with the (old) custom heavy ion reconstruction sequence.

@mandrenguyen

### CODE REVIEW

**Comment by slava77** [comment]
On 5/31/18 7:58 AM, bi-ran wrote:
> this PR includes the following modifications to electron reconstruction
> for the 2018 heavy ion era:
> 
>   * raise supercluster energy thresholds
>   * removal of tracker-driven electrons
> 
> these changes affect mainly very low-pt electrons, and are in line with
> the (old) custom heavy ion reconstruction sequence.
> 
> @mandrenguyen <https://github.com/mandrenguyen>

please provide a link to some slides (in an HI meeting?) that describe
the observed or expected performance changes, including also the
estimates of the CPU time changes

Thank you.

File: RecoEgamma/EgammaElectronProducers/python/ecalDrivenElectronSeedsParameters_cff.py [hunk 1]
```diff
@@ -81,3 +81,5 @@
     
 )
 
+from Configuration.Eras.Modifier_pp_on_AA_2018_cff import pp_on_AA_2018
+pp_on_AA_2018.toModify(ecalDrivenElectronSeedsParameters, SCEtCut = cms.double(15.0))
```
Comment (slava77) [inline RecoEgamma/EgammaElectronProducers/python/ecalDrivenElectronSeedsParameters_cff.py]: please drop the types in all modifications.
This is a safer use of syntax against typos and also helps with possible future parameter name changes and migrations

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/23404	f48840f
- code changes are in line with the PR description and more clearly so with the description of this PR in the RECO meeting on June 1 ("Move SC thresholds to 15 GeV, and remove tracker-driven electrons"). The changes affect only Run2_2018_pp_on_AA
- jenkins tests pass
- local tests with 200 events of wf 158.0 (uses Run2_2018_pp_on_AA) show expected behavior. 
    - RECO time went down from 9.0 s/ev to 8.9 s/ev primarily related to the modified electron reco modules (the related reco modules were not particularly costly to start with).
    - AOD size reduces ~negligibly.

**Comment by fabiocos** [comment]
+1

**Pull request merged**